### PR TITLE
More capella fixes

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
follow up to #23654, now all `Q_ASSERT()` got converted into `IF_ASSERT_FAILED()` and do a `throw Capella::Error::BAD_FORMAT;` in non-debug builds

### CODE REVIEW

File: src/importexport/capella/internal/capella.cpp [hunk 1]
```diff
@@ -2352,7 +2358,7 @@ void Capella::readLayout()
 
     unsigned char b  = readByte();
     IF_ASSERT_FAILED((b & 0xFC) == 0) {   // bits 2...7 reserviert
-        //throw Capella::Error::BAD_FORMAT;
+        throw Capella::Error::BAD_FORMAT;
```
Comment (Jojo-Schmitz) [inline src/importexport/capella/internal/capella.cpp:2361]: this one was commented out by mistake

File: src/importexport/capella/internal/capella.cpp [hunk 2]
```diff
@@ -1548,7 +1548,9 @@ void TransposableObj::read()
     if (variants.size() != b) {
         LOGD("variants.size %lld, expected %d", variants.size(), b);
     }
-    Q_ASSERT(variants.size() == b);
+    IF_ASSERT_FAILED(variants.size() == b) {
```
Comment (cbjeukendrup) [inline src/importexport/capella/internal/capella.cpp:1551]: This one can be combined with the `if` just above it

File: src/importexport/capella/internal/capella.cpp [hunk 3]
```diff
@@ -1632,7 +1634,9 @@ void NotelinesObj::read()
     case 1: break;         // Einlinienzeile
     case 2: break;         // Standard (5 Linien)
     default: {
-        Q_ASSERT(b == 0);
+        IF_ASSERT_FAILED(!b) {
```
Comment (cbjeukendrup) [inline src/importexport/capella/internal/capella.cpp:1637]: I think `== 0` conveys the intention slightly clearer

File: src/importexport/capella/internal/capella.cpp [hunk 4]
```diff
@@ -2351,8 +2357,8 @@ void Capella::readLayout()
     adjustVert = readByte();      // 0=nein, 1=außer letzte Seite, 3=alle Seiten
 
     unsigned char b  = readByte();
-    IF_ASSERT_FAILED((b & 0xFC) == 0) {   // bits 2...7 reserviert
-        //throw Capella::Error::BAD_FORMAT;
+    IF_ASSERT_FAILED(!(b & 0xFC)) {   // bits 2...7 reserviert
```
Comment (cbjeukendrup) [inline src/importexport/capella/internal/capella.cpp:2360]: (but this change is fine)

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**